src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EnvironmentVariableActionTests.cs

```diff
@@ -0,0 +1,219 @@
+        {
+            _outputHelper = outputHelper;
+            _endpointUtilities = new(_outputHelper);
+        }
+
+        /// <summary>
+        /// Test that the <see cref="SetEnvironmentVariableActionFactory.SetEnvironmentVariableAction"/> works correctly.
+        /// </summary>
+        /// <remarks>
+        /// The required APIs only exist on .NET 6.0+
+        /// </remarks>
+        [Theory]
+        [MemberData(nameof(ActionTestsHelper.Get6PlusTfms), MemberType = typeof(ActionTestsHelper))]
+        public async Task TestSetEnvVar(TargetFrameworkMoniker tfm)
+        {
+            await TestHostHelper.CreateCollectionRulesHost(
+                outputHelper: _outputHelper,
+                setup: (Tools.Monitor.RootOptions rootOptions) =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(DefaultVarName, DefaultVarValue);
+                },
+                hostCallback: async (Extensions.Hosting.IHost host) =>
+                {
+                    SetEnvironmentVariableOptions setOpts = ActionTestsHelper.GetActionOptions<SetEnvironmentVariableOptions>(host, DefaultRuleName, 0);
+
+                    ICollectionRuleActionOperations actionOperationsService = host.Services.GetService<ICollectionRuleActionOperations>();
+                    Assert.True(actionOperationsService.TryCreateFactory(KnownCollectionRuleActions.SetEnvironmentVariable, out ICollectionRuleActionFactoryProxy setFactory));
+
+                    EndpointInfoSourceCallback endpointInfoCallback = new(_outputHelper);
+                    await using ServerSourceHolder sourceHolder = await _endpointUtilities.StartServerAsync(endpointInfoCallback);
+
+                    AppRunner runner = _endpointUtilities.CreateAppRunner(sourceHolder.TransportName, tfm);
+                    runner.ScenarioName = TestAppScenarios.EnvironmentVariables.Name;
+
+                    Task<IEndpointInfo> newEndpointInfoTask = endpointInfoCallback.WaitAddedEndpointInfoAsync(runner, CommonTestTimeouts.StartProcess);
+
+                    await runner.ExecuteAsync(async () =>
+                    {
+                        IEndpointInfo endpointInfo = await newEndpointInfoTask;
+
+                        ICollectionRuleAction setAction = setFactory.Create(endpointInfo, setOpts);
+
+                        using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(CommonTestTimeouts.EnvVarsTimeout);
+
+                        await ActionTestsHelper.ExecuteAndDisposeAsync(setAction, CommonTestTimeouts.EnvVarsTimeout);
+
+                        using CancellationTokenSource waitTimeout = new CancellationTokenSource(CommonTestTimeouts.SendCommand);
+                        Task<string> getSetVar = runner.WaitForEnvironmentVariable(DefaultVarName, waitTimeout.Token);
```

Comment:
Add an extension method to AppRunnerExtensions that takes the environment variable name and uses the default timeout so that the test can be simplified a little.

Consider wrapping all of the WaitForEnvironmentVariable, SendCommandAsync, and awaited value into a new extension method that does all of that since this seems to be repeated:

```cs
public static async Task<string> GetEnvironmentVariable(this AppRunner runner, string name)
{
    Task<string> getVarTask = runner.WaitForEnvironmentVariable(name);
    await runner.SendCommandAsync(TestAppScenarios.EnvironmentVariables.Commands.PrintVars);
    return getVarTask;
}
```